### PR TITLE
Feature/internationalization

### PR DESCRIPTION
**Pull request by yashkapoor8710**
Simulator Screen Shot - iPhone 11 - 2020-03-23 at 12 56 07

### CODE REVIEW

**Comment by yashkapoor8710** [comment]
Screenshot 2020-03-23 at 1 15 07 PM

File: components/Chat/languageSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+const ThemeToggle: React.FunctionComponent<Props> = ({
+  updateLanguage,
+}: Props) => {
+  const theme: AppTheme = useTheme();
+  const { selectedLanguage }: AppConstants = useConstants();
+  const [language, setLanguage] = useState<string>(selectedLanguage);
+
+  console.log(selectedLanguage , "check")
```
Comment (manojsinghnegiwd) [inline components/Chat/languageSelector.tsx]: remove console.log

File: components/Chat/languageSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  const languages = [
+    { label: 'English', value: 'English' },
+    { label: 'French', value: 'French' },
+    { label: 'Spanish', value: 'Spanish' },
+    { label: 'German', value: 'German' },
+    { label: 'Chinese', value: 'Chinese' },
+  ];
+
+  useEffect(() => {
+
```
Comment (manojsinghnegiwd) [inline components/Chat/languageSelector.tsx]: remove this spaces

File: components/Chat/languageSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    updateLanguage(language)
+  }, []);
+
+  const onChangeLanguage = (value) => {
+    updateLanguage(value)
+    setLanguage(value)
+  }
+
+  console.log(language , "lang")
```
Comment (manojsinghnegiwd) [inline components/Chat/languageSelector.tsx]: remove console.log

File: config/languages.ts [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        username: `Username is required`,
+        email: `Email is required`,
+        validEmail: 'Please enter a valid email',
+        password: `Password is required`,
+        passwordLength: 'Password length between 6 and 12',
+      },
+}
+
+export const chinese: AppConstants = {
+    selectedTheme: ThemeKey.light,
```
Comment (manojsinghnegiwd) [inline config/languages.ts]: You don't need this.

File: config/languages.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    selectedLanguage: LanguageKey.chinese,
```
Comment (manojsinghnegiwd) [inline config/languages.ts]: Or this.

File: components/Chat/languageSelector.tsx (hunk 1 above)
Comment (yashkapoor8710) [reply components/Chat/languageSelector.tsx]: I have removed this

File: components/Chat/languageSelector.tsx (hunk 3 above)
Comment (yashkapoor8710) [reply components/Chat/languageSelector.tsx]: I have removed this

File: components/Chat/languageSelector.tsx (hunk 2 above)
Comment (yashkapoor8710) [reply components/Chat/languageSelector.tsx]: I have removed this

File: config/languages.ts (hunk 4 above)
Comment (yashkapoor8710) [reply config/languages.ts]: I have removed this

File: config/languages.ts (hunk 5 above)
Comment (yashkapoor8710) [reply config/languages.ts]: I have removed this

File: store/reducers/config.ts [hunk 6]
```diff
@@ -6,6 +6,10 @@ enum ThemeActionType {
     SET_THEME = "SET_THEME"
 }
 
+enum ThemeActionType {
+    SET_LANGUAGE = "SET_LANGUAGE"
```
Comment (manojsinghnegiwd) [inline store/reducers/config.ts]: Define it with `SET_THEME`

File: pages/Signup/index.tsx [hunk 7]
```diff
@@ -17,46 +18,47 @@ interface LoginField {
...
-    
-    return errors
-}
-
 const Signup: React.FunctionComponent<RouteComponentProps> = ({
     history
 }: RouteComponentProps) => {
 
     const theme: AppTheme = useTheme();
+    const constants: AppLanguage = useLanguage();
```
Comment (manojsinghnegiwd) [inline pages/Signup/index.tsx]: Rename this to `Language`

File: pages/Login/index.tsx [hunk 8]
```diff
@@ -1,47 +1,51 @@
...
+                    max: 12,
+                    errorMsg: constants.loginValidation.passwordLength
+                }
+            },
+        },data)
+        
+        return errors
+    }
+
     useEffect(() => {
```
Comment (manojsinghnegiwd) [inline pages/Login/index.tsx]: This useEffect should have a empty array dependency.

Comment (yashkapoor8710) [reply pages/Login/index.tsx]: I have added an empty array dependency.

File: store/reducers/config.ts (hunk 6 above)
Comment (yashkapoor8710) [reply store/reducers/config.ts]: I have defined them together

File: pages/Signup/index.tsx (hunk 7 above)
Comment (yashkapoor8710) [reply pages/Signup/index.tsx]: I have renamed this

**Pull request merged**